include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -36,15 +36,17 @@ __set_active_device(sycl::device* __new_active_device)
     __active_device.store(__new_active_device, std::memory_order_release);
 }
 
-static sycl::device
-__get_offload_device()
+using __device_selector_type = int (*)(const sycl::device &);
+
+static __device_selector_type
```

Comment:
```suggestion
static auto
```

If the constructor would be `template` - `__device_selector_type` can be removed

Reply:
Done.